packages/scoutgame/src/protocol/generateWeeklyClaims.ts

```diff
@@ -168,51 +43,71 @@ export async function generateWeeklyClaims({
     throw new Error(`Claims for week ${week} already exist`);
   }
 
-  const { claims, claimsWithUserId, builderEvents, tokenReceipts, weeklyClaimId } = await calculateWeeklyClaims({
-    week
-  });
-
-  const { rootHash, tree } = generateMerkleTree(claims);
+  const { claims, builderEvents, tokenReceipts, weeklyClaimId } = weeklyClaimsCalculated;
 
-  const proofsMap: Record<string, string[]> = {};
+  const proofsMap: Record<Address, string[]> = {};
 
-  for (const claim of claimsWithUserId) {
-    const proof = getMerkleProofs(tree, { address: claim.address, amount: claim.amount });
+  for (const claim of claims) {
+    const proof = getMerkleProofs(weeklyClaimsCalculated.merkleProofs.tree, {
+      address: claim.address,
+      amount: claim.amount
+    });
 
-    proofsMap[claim.userId] = proof;
+    proofsMap[claim.address] = proof;
   }
 
+  const rootHashWithNullByte = `0x${weeklyClaimsCalculated.merkleProofs.rootHash}`;
+
   const claimsBody: ClaimsBody = {
-    leaves: claims,
-    leavesWithUserId: claimsWithUserId
+    leaves: claims
   };
 
-  await protocolImplementationWriteClient().setMerkleRoot({
+  await protocolImplementationWriteClient().setWeeklyMerkleRoot({
     args: {
-      merkleRoot: `0x${rootHash}`,
-      week
+      weeklyRoot: {
+        isoWeek: week,
+        // Tokens can be claimed until the end of the season and the next season
+        validUntil: Math.round(getEndOfSeason(currentSeason).plus({ weeks: 13 }).toSeconds()),
+        merkleRoot: rootHashWithNullByte,
+        // Stub definition until we add in IPFS
+        merkleTreeUri: `ipfs://scoutgame/merkle-tree/${week}`
+      }
     }
   });
 
-  const [weeklyClaim] = await prisma.$transaction([
-    prisma.weeklyClaims.create({
+  // Fund the protocol contract
+  await getSablierLockupContract()
+    .claim({
+      args: {
+        streamId: BigInt(sablierStreamId)
+      }
+    })
+    .catch((error: any) => {
```

Comment:
this means teh contract isn't getting tokens. I believe it needs to blow up, not fail silently